IDinsight/aaq-core
# For each guardrail, add separate env var for LLM model and endpoint

Reviewer: @suzinyou 
Estimate: 30mins

---

## Ticket

Fixes: [396](https://idinsight.atlassian.net/browse/AAQ-396)

## Description, Motivation and Context

### Goal

- Allow us to select a different LLM model and endpoint for each guardrail step. This will help us save costs by not calling GPT-4 for everything.

### Changes

- Added new environment variables that will be used to determine which LiteLLM model and endpoint to use for each call of the `_ask_llm_async()` function.
  - By default, every LLM model and endpoint will be set to the `LITELLM_MODEL_DEFAULT` and `LITELLM_ENDPOINT_DEFAULT` values unless specified.
- Update default model to gpt-4-0125-preview from 1106

## How has this been tested?

- Set different models and endpoints as environment variables (see `core_backend/app/config.py`) and run the backend. When you send a request to the `/llm-response` or `/embeddings-search` endpoints, you will see that different models are being used by LiteLLM in the terminal logs.
- Run the docker-compose deployment (see updated `template.env`) and also test functionality.

## Checklist

Fill with `x` for completed. Delete any lines that are not relevant

- [ x I have reviewed my own code to ensure good quality
